warn/scrapers/la.py

```diff
@@ -0,0 +1,95 @@
+import os
+import logging
+import re
+from pathlib import Path
+
+import pdfplumber
+from bs4 import BeautifulSoup
+
+from .. import utils
+from ..cache import Cache
+
+__authors__ = ["chriszs"]
+__tags__ = ["pdf"]
+
+logger = logging.getLogger(__name__)
+
+
+def scrape(
+    data_dir: Path = utils.WARN_DATA_DIR,
+    cache_dir: Path = utils.WARN_CACHE_DIR,
+) -> Path:
+    """
+    Scrape data from Lousiana.
+
+    Keyword arguments:
+    data_dir -- the Path were the result will be saved (default WARN_DATA_DIR)
+    cache_dir -- the Path where results can be cached (default WARN_CACHE_DIR)
+
+    Returns: the Path where the file is written
+    """
+    cache = Cache(cache_dir)
+
+    state_code = "la"
+    base_url = "https://www.laworks.net/"
+    url = f"{base_url}Downloads/Downloads_WFD.asp"
+
+    cache_key = f"{state_code}/Downloads_WFD.html"
+
+    page = utils.get_url(url)
+    html = page.text
+
+    cache.write(cache_key, html)
+
+    document = BeautifulSoup(html, "html.parser")
+    links = document.find_all("a")
+
+    pdf_urls = [
+        f"{base_url}{link['href']}" for link in links if "WARN Notices" in link.text
+    ]
+
+    output_rows = []
+
+    for pdf_index, pdf_url in enumerate(pdf_urls):
+        file_name = os.path.basename(pdf_url)
+        cache_key = f"{state_code}/{file_name}"
+
+        pdf_path = cache.download(cache_key, pdf_url)
+
+        with pdfplumber.open(pdf_path) as pdf:
+            for page_index, page in enumerate(pdf.pages):
+                rows = page.extract_table()
+
+                for row_index, row in enumerate(rows):
+                    output_row = [_clean_text(cell) for cell in row]
+
+                    # Skip headers on all but first page of first PDF
+                    if (pdf_index == 0 and page_index == 0) or row_index != 0:
+                        output_rows.append(output_row)
```

Comment:
```suggestion
                # Loop through the rows
                for row_index, row in enumerate(rows):
                    # Skip headers on all but first page of first PDF
                    if pdf_index > 0 and row_index == 0:
                        print(row)
                        logger.debug(f"Skipping header row on PDF {pdf_index+1} page {page_index+1}")
                        continue

                    # Extract data
                    output_row = [_clean_text(cell) for cell in row]

                    # Write row
                    output_rows.append(output_row)
```
This doesn't solve your "sometimes there a header, and sometimes there isn't a header` problem, but it might set up a clearer conditional clause where you can experiment with the fix.